kong/llm/drivers/cohere.lua

```diff
@@ -12,102 +12,66 @@ local table_new = require("table.new")
-      request_table.message = request_table.messages[#request_table.messages].content
-      request_table.messages = nil
-
-    elseif request_table.prompt then
-      request_table.temperature = model.options.temperature
-      request_table.max_tokens = model.options.max_tokens
-      request_table.truncate = request_table.truncate or "END"
-      request_table.return_likelihoods = request_table.return_likelihoods or "NONE"
-      request_table.p = model.options.top_p
-      request_table.k = model.options.top_k
-
+      request_table.chat_history = chat_history
     end
 
-    return request_table, "application/json", nil
-  end,
-
-  ["llm/v1/completions"] = function(request_table, model)
-    request_table.model = model.name
-
-    if request_table.prompt and request_table.messages then
-      return kong.response.exit(400, "cannot run a 'prompt' and a history of 'messages' at the same time - refer to schema")
-
-    elseif request_table.messages then
-      -- we have to move all BUT THE LAST message into "chat_history" array
-      -- and move the LAST message (from 'user') into "message" string
-      if #request_table.messages > 1 then
-        local chat_history = table_new(#request_table.messages - 1, 0)
-        for i, v in ipairs(request_table.messages) do
-          -- if this is the last message prompt, don't add to history
-          if i < #request_table.messages then
-            local role
-            if v.role == "assistant" or v.role == "CHATBOT" then
-              role = "CHATBOT"
-            else
-              role = "USER"
-            end
-    
-            chat_history[i] = {
-              role = role,
-              message = v.content,
-            }
-          end
-        end
-
-        request_table.chat_history = chat_history
-      end
+    request_table.max_tokens = request_table.max_tokens or model.options.max_tokens
+    request_table.temperature = request_table.temperature or model.options.temperature
+    request_table.truncate = request_table.truncate or "END"
+    request_table.return_likelihoods = request_table.return_likelihoods or "NONE"
+    request_table.p = request_table.top_p or model.options.top_p
+    request_table.k = request_table.top_k or model.options.top_k
+    request_table.message = request_table.messages[#request_table.messages].content
+    request_table.messages = nil
+
+  elseif request_table.prompt then
+    request_table.prompt = request_table.prompt
+    request_table.messages = nil
+    request_table.message = nil
+    request_table.temperature = request_table.temperature or model.options.temperature
+    request_table.max_tokens = request_table.max_tokens or model.options.max_tokens
+    request_table.truncate = request_table.truncate or "END"
+    request_table.return_likelihoods = request_table.return_likelihoods or "NONE"
+    request_table.p = request_table.top_p or model.options.top_p
+    request_table.k = request_table.top_k or model.options.top_k
```

Comment:
Looks like some of these values are set the same way in both branches (e.g. `request_table.p`). Consider moving the common bits outside of the conditional unless you think it'll make the code uglier/less readable.